job-server/src/spark.jobserver/WebApi.scala

```diff
@@ -504,6 +536,43 @@ class WebApi(system: ActorSystem,
     }
   }
 
+  private def sendStreamingResponse(ctx: RequestContext,
+                                    chunkSize: Int,
+                                    byteIterator: Iterator[_]): Unit = {
+    // simple case class whose instances we use as send confirmation message for streaming chunks
+    case class Ok(remaining: Iterator[_])
```

Comment:
I think this entire Actor definition should be moved out to another file/Class by itself, instead of being defined inline.   Would be more readable.  Perhaps sendStreamingResponse should be moved out to a trait as well.